### ISSUE TITLE
FP - Update Apt Get Install Lists Were Not Deleted query for Dockerfile

### ISSUE THREAD

**Issue by jonny-wg2** [Checkmarx/kics#3762]
### Platform
Dockerfile

### Provider
Dockerfile

### Description
The rule gives a false positive as the following below is correct according to the Docker documentation. https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments

`Dockerfile`
```
RUN apt-get update && apt-get install -y --no-install-recommends  \
    bash \
    curl \
    git \
    vim \
    jq \
    gzip \
    python3 \
    python3-pip \
    expect \
    && rm -rf /var/lib/apt/lists/*
```

`results.json`
```
{
	"files_scanned": 1,
	"files_parsed": 1,
	"files_failed_to_scan": 0,
	"queries_total": 39,
	"queries_failed_to_execute": 0,
	"queries_failed_to_compute_similarity_id": 0,
	"queries": [
		{
			"query_name": "Apt Get Install Lists Were Not Deleted",
			"query_id": "df746b39-6564-4fed-bf85-e9c44382303c",
			"query_url": "https://docs.docker.com/develop/develop-images/dockerfile_best-practices/",
			"severity": "INFO",
			"platform": "Dockerfile",
			"files": [
				{
					"file_name": "/path/Dockerfile",
					"similarity_id": "08b2360e177aeacd869858a0b6957ea292d68640e25f09f51bd1bc6ed96ddbf5",
					"line": 4,
					"issue_type": "IncorrectValue",
					"search_key": "FROM={{registry.wgtwo.com/docker/library/ubuntu:21.04}}.RUN={{apt-get update \u0026\u0026 apt-get install -y --no-install-recommends      bash     curl     git     vim     jq     gzip     python3     python3-pip     expect     \u0026\u0026 rm -rf /var/lib/apt/lists/*}}",
					"search_value": "",
					"expected_value": "After using apt-get install, it is needed to delete apt-get lists",
					"actual_value": "After using apt-get install, the apt-get lists were not deleted",
					"value": null
				}
			],
			"category": "Supply-Chain",
			"description": "After using apt-get install, it is needed to delete apt-get lists"
		}
	],
	"scan_id": "console",
	"severity_counters": {
		"HIGH": 0,
		"INFO": 1,
		"LOW": 0,
		"MEDIUM": 0
	},
	"total_counter": 1,
	"start": "2021-06-28T08:46:47.3493291Z",
	"end": "2021-06-28T08:46:48.1578587Z",
	"paths": [
		"/path"
	]
}
```

**Comment by cosmicgirl97** [comment]
The query requires `apt-get clean` before `rm -rf /var/lib/apt/lists/*`. That's why it is reporting a result (the sample does not define it). 
According to the [Docker documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run), 

> Official Debian and Ubuntu images automatically run apt-get clean, so explicit invocation is not required.

Maybe `apt-get clean` check can be deleted. What do you think? @jonny-wg2 @felipe-avelar @rogeriopeixotocx @joaoReigota1

**Comment by rogeriopeixotocx** [comment]
Hello,

Thank you for the feedback.

I believe was already fixed by https://github.com/Checkmarx/kics/pull/3721

In my local tests, the provided Dockerfile is not triggering the query "Apt Get Install Lists Were Not Deleted" query on the latest KICS version.

Currently, the query checks for [apt-get clean OR rm -rf](https://github.com/Checkmarx/kics/blob/master/assets/queries/dockerfile/apt_get_install_lists_were_not_deleted/query.rego#L22-L36).

@cosmicgirl97 Just by looking at the `FROM` statement, I think the only way we can tell if the current `Dockerfile` is referring to the official Debian or Ubuntu images is if it uses the `latest` tag. Anything different than this could be just another local retagged docker image. We could add this validation to turn off the query if a `FROM ubuntu`/`FROM ubuntu:latest`/`FROM debian:latest` is found in the current scope. But it sounds like an edge case, the user can always turn off this query or exclude the result.

@jonny-wg2 are you sure you're using our latest KICS version?

**Comment by jonny-wg2** [comment]
Gotcha, thanks for all of the useful feedback.

I did not have `apt-get clean`

I was running kics v1.3.3

**Comment by rogeriopeixotocx** [comment]
Thank you, closing this
